documentation/internal/Filters-for-Applications.md

```diff
@@ -0,0 +1,292 @@
+
+The `type` is the template code for the applications type, also displayed in the header of the Applications list.
+The `user-role` is used to build how the columns and rows should be displayed. The user-role is deduced from the permissions that the current user have to view this application type. In this example the user has **applicant** role by what was set in their permissions.
+
+### Example: [UI Design] Applicant - applications list
+
+![Applicant-List](images/Filters-for-Applications-Applicant-List.png)
+
+---
+
+### Filters
+
+---
+
+#### Label filters:
+
+##### type
+
+**This is a compulsory filter**
+In case none is included in URL will be deduced from 1st user's persmissions
+
+Example: `type=user-registration`
+Considered in implementation: **Yes** [#131 - Headers]
+Can have combined values: **No**
+
+Options:
+
+- `template-code` - Of application type in list (and header)
+
+##### category
+
+Example: `category=company`
+Considered in implementation: **Yes** **[New issue?]**
+Can have combined values: **No**
+
+Options:
+
+- `category-code` - Group of templates (application types)
+
+##### stage
+
+Example: `stage=screening`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `stage=screening,assessment,final-decision`
+
+Options:
+
+- Set of `stage-name` which is deduced per template (application type)
+
+##### sort-by
+
+Example: `sort-by=stage:asc`
+Considered in implementation: **Yes** [#131 - Headers]
+Can have combined values: **Yes** `sort-by=stage:asc,application-name:desc`
+
+Options:
+
+- `column-name:direction`
+- direction [Optional] `asc`/`desc`
+- if no direction is included use default: `asc`
+
+##### user-role
+
+**This is a **compulsory** filter.**
+In case none is included in URL will be deduced from permission of type in user's persmissions.
+
+Example: `user-role=applicant`
+Considered in implementation: **Yes** [#131 - Headers]
+Can have combined values: **No**
+
+Options:
+
+- Deduced by user's permissions (**TODO: More detailed explanation of user-role per permissions**)
+
+##### outcome
+
+Example: `outcome=pending`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `outcome=pending,approved`
+
+Options:
+
+- Set of `outcome` common for all templates (application types)
+
+##### action
+
+Example: `action=edit-draft`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `action=edit-draft,`
+
+Options:
+
+- Applicant:
+  `edit-draft`
+  `make-updates`
+  `renew`
+  `view` (Submitted)
+- Reviewer
+  `start-review`
+  `continue-review`
+  `review-updates`
+  `view` (Submitted)
+- Consolidator
+  `assign`
+  `consolidate`
+  `review-updates`
+  `view` (Submitted)
```

Comment:
I mentioned this in the doc too, but I don't really understand how "actions" can be filters. Surely actions are links of things you can do, and are determined by your permissions and role?

Reply:
Yes, actions are a set of possible things that each user-role can do, but I was thinking it would be useful to search and see only applications that you can `make-updates` for example. I know this is the same as a filter that would search for application with status == `Changes required` but since this is not visible in the UI I thought using actions would be a better filter for users. Note that there isn't one status filter for the same reason.